### PR TITLE
has-armor-type sexp

### LINKED ISSUE #2854: FR: a SEXP to check the currently applied armor of an object
Just as is say, I would like to request the inclusion of a SEXP that lets you check which armor class is currently applied to a ship, please.

This is for gameplay effects that use dynamic hull and shield armor changes, e.g. entering and leaving field effects that interfere with shields or active armor (buff or debuff area of effect basically).
Currently you have to work around and use a ship flag as proxy check.

### PR DESCRIPTION
**Pull request by MjnMixael**
Adds the sexp to check the current armor of a ship or ship's subsystem. Fixes #2854

### CODE REVIEW

**Review by Goober5000 — CHANGES_REQUESTED** [review]

File: code/parse/sexp.cpp [hunk 1]
```diff
@@ -25197,6 +25198,52 @@ int sexp_is_in_mission(int node)
...
+			if (ss == nullptr) {
+				continue;
+			}
+
+			currentArmor = ss->armor_type_idx;
+		}
+	}
+
+	if (currentArmor == armor)
+		return SEXP_TRUE;
```
[the remark this answers is not in the document]
Comment (MjnMixael) [reply code/parse/sexp.cpp]: done

File: code/parse/sexp.cpp [hunk 2]
```diff
@@ -25197,6 +25198,59 @@ int sexp_is_in_mission(int node)
...
+	// get armor from sexp
+	int armor;
+	if (!stricmp(SEXP_NONE_STRING, CTEXT(node))) {
+		armor = -1;
+	} else {
+		armor = armor_type_get_idx(CTEXT(node));
+
+		//warn and return if the armor is not found
+		if (armor < 0) {
+			Warning(LOCATION, "Armor %s not found on ship %s", CTEXT(node), shipp->ship_name);
```
Comment (Goober5000) [inline code/parse/sexp.cpp:25223]: The warning message is not quite correct... the armor is not recognized as an armor type.  Yes, it's not found on the ship, but more fundamentally, it's not a valid entry in the armor table.

**Review by Goober5000 — CHANGES_REQUESTED** [review]

File: code/parse/sexp.cpp (hunk 2 above)
Comment (MjnMixael) [reply code/parse/sexp.cpp:25223]: done

**Review by Goober5000 — APPROVED** [review]

**Comment by Goober5000** [comment]
0rph3u5 expressed his appreciation:

> please relay my thanks to everyone involved 👍

**Pull request merged**